src/config/jest-config-resolver.ts

```diff
@@ -0,0 +1,48 @@
+import * as fs from 'fs'
+import * as path from 'path'
+import * as resolve from 'resolve'
+
+import { Logger } from 'bs-logger'
+
+interface IPackageJson {
+  main: string
+}
+
+export function getJestConfigPkg<TJestConfig>(logger: Logger): TJestConfig {
+  try {
+    const jestPath: string = resolvePackagePath('jest', __dirname)
+    const jestCliPath: string = resolvePackagePath('jest-cli', jestPath)
+    const jestConfigPath: string = resolvePackagePath('jest-config', jestCliPath)
+
+    const jestConfigPackageJson: IPackageJson = require(path.join(jestConfigPath, 'package.json'))
+    const jestConfigMainPath: string = path.resolve(jestConfigPath, jestConfigPackageJson.main)
+    return require(jestConfigMainPath)
+  } catch (e) {
+    logger.error('Unable to resolve jest-config. Ensure Jest is properly installed.')
```

Comment:
I believe you can prefix here the error in the context: `logger.error({error}, ...)

Also we try to centralize all error messages in the `messages.ts`, if you can do so as well that would be nice.